python/lsst/daf/butler/_query.py

```diff
@@ -0,0 +1,267 @@
+            which a dataset of that dataset type appears (according to the
+            order of ``collections`` passed in).  If `True`, ``collections``
+            must not contain regular expressions and may not be ``...``.
+        data_id : `dict` or `DataCoordinate`, optional
+            A data ID whose key-value pairs are used as equality constraints
+            in the query.
+        where : `str`, optional
+            A string expression similar to a SQL WHERE clause.  May involve
+            any column of a dimension table or (as a shortcut for the primary
+            key column of a dimension table) dimension name.  See
+            :ref:`daf_butler_dimension_expressions` for more information.
+        bind : `~collections.abc.Mapping`, optional
+            Mapping containing literal values that should be injected into the
+            ``where`` expression, keyed by the identifiers they replace.
+            Values of collection type can be expanded in some cases; see
+            :ref:`daf_butler_dimension_expressions_identifiers` for more
+            information.
+        **kwargs
+            Additional keyword arguments are forwarded to
+            `DataCoordinate.standardize` when processing the ``dataId``
+            argument (and may be used to provide a constraining data ID even
+            when the ``dataId`` argument is `None`).
+
+        Returns
+        -------
+        refs : `.queries.DatasetQueryResults`
+            Dataset references matching the given query criteria.  Nested data
+            IDs are guaranteed to include values for all implied dimensions
+            (i.e. `DataCoordinate.hasFull` will return `True`), but will not
+            include dimension records (`DataCoordinate.hasRecords` will be
+            `False`) unless `~.queries.DatasetQueryResults.expanded` is
+            called on the result object (which returns a new one).
+
+        Raises
+        ------
+        lsst.daf.butler.registry.DatasetTypeExpressionError
+            Raised when ``datasetType`` expression is invalid.
+        TypeError
+            Raised when the arguments are incompatible, such as when a
+            collection wildcard is passed when ``findFirst`` is `True`, or
+            when ``collections`` is `None` and ``self.defaults.collections`` is
+            also `None`.
+        lsst.daf.butler.registry.DataIdError
+            Raised when ``dataId`` or keyword arguments specify unknown
+            dimensions or values, or when they contain inconsistent values.
+        lsst.daf.butler.registry.UserExpressionError
+            Raised when ``where`` expression is invalid.
+
+        Notes
+        -----
+        When multiple dataset types are queried in a single call, the
+        results of this operation are equivalent to querying for each dataset
+        type separately in turn, and no information about the relationships
+        between datasets of different types is included.  In contexts where
```

Comment:
Let's drop the rest of the note after `included.`, since it references `Registry` methods that are not going to remain the preferred way of doing things, and the new way has fully materialized yet.